mathics/builtin/numbers/constants.py

```diff
@@ -307,15 +265,15 @@ def apply_N(self, precision, evaluation, options={}):
             return PrecisionReal((sympy.pi / 180).n(d))
 
 
-class E(_MPMathConstant, _NumpyConstant, _SympyConstant):
+class E(_MPMathConstant, _SympyConstant):
     """
     <dl>
     <dt>'E'</dt>
         <dd>is the constant \u2107 with numerical value \u2243 2.71828.
     </dl>
 
     >> N[E]
-     = 2.71828
+     = 2.71828182845905
```

Comment:
See above. This is doesn't follow the semantics of WL.

Reply:
@rocky should I revert it?

Reply:
I believe so.  When using mpmath we need to figure out how to make it match https://reference.wolfram.com/language/guide/PrecisionAndAccuracyControl.html 

Somewhere in old tickets this has come up before. @mmatera probably knows more about this than I do. 

Writing some sort of compatibility layer on top of mpmath in a lower-level language is one areas where the use of Cython, C, Rust, etc. may be  justified. 

However you should think of that akin to a project of its own like graphics3d (although probably not as extensive or as hard.)

Reply:
> Writing some sort of compatibility layer on top of mpmath in a lower-level language is one areas where the use of Cython, C, Rust, etc. may be justified.

I guess Python is the best here, mpmath is written in Python.

Reply:
> I believe so.

Ok. I'll remove it.